db.go

```diff
@@ -411,6 +457,16 @@ func (db *DB) Close() error {
 			return err
 		}
 	}
+
+	// If no persistence is enabled; we're done
+	if db.bucket == nil {
+		return nil
+	}
+
+	// Persist blocks to storage
```

Comment:
I like where we're going with this, but it needs more thought I think. Essentially we're getting rid of all state at shutdown here, which is amazing! But that also means that if this is successful we can (and should) drop the write-ahead log if it's enabled, otherwise, we are very likely to duplicate all data in the WAL into object storage causing ourselves a ton of unnecessary deduplication work.

Additionally (this is more a follow-up), we should coordinate with the insertion layer that on shutdown we stop accepting writes, then wait for all active writers to finish, and then perform all of this. That way we maintain the highest chance for correctness and durability.

Reply:
Addressed dropping the WAL upon successful closing of tables